### PR TITLE
Fix vision

### LINKED ISSUE #24: Fix vision
Calls to GPT 4 vision are performed outside of a function calling loop because GPT 4 vision doesn't support function calling and I was too lazy to do this properly. 

I think we can get around this by implementing vision analysis as a function/tool that is invocable within a FC loop.  When an image is provided in the prompt request, we can force tool selection, which runs a separate prompt thread for image analysis.

### PR DESCRIPTION
**Pull request by skylar-anderson**
closes #24

### CODE REVIEW

**Comment by skylar-anderson** [comment]
This is super exciting! Can't wait to try it

**Comment by skylar-anderson** [comment]
I'm concerned that the max token limit may be too large for the vision prompt. Have you tested it?

**Comment by skylar-anderson** [comment]
LGTM, but I have some minor nits I can share with you in slack.

**Comment by skylar-anderson** [comment]
## Pull Request Digest 

### :sparkles: TL;DR
- Implements the `analyzeImage` function utilizing GPT-4 Vision.
- Closes issue #24 for enhanced image analysis capabilities.
- Community excitement evident, with minor concerns raised about token limits.
- Introduces significant code changes, including the addition of new files and logic for image analysis.
- Invites further discussion of minor issues out of band on Slack.

### :bulb: Intent of the PR
The intent behind this pull request is to fix issues with the vision component of the project, specifically by integrating a new `analyzeImage` function. This implementation aims to leverage GPT-4 Vision to analyze images and respond to inquiries about them, thereby closing issue #24.

### :speech_balloon: Community Feedback
One comment exudes excitement:
> "This is super exciting! Can't wait to try it"

Meanwhile, another raises a valid concern:
> "I'm concerned that the max token limit may be too large for the vision prompt. Have you tested it?"

Additionally, a comment suggests that there are some minor adjustments to discuss, but these are likely not blockers for the PR:
> "LGTM, but I have some minor nits I can share with you in Slack."

### :mag: Walkthrough of Changes
The PR adds a new `.vscode/launch.json` file enabling debugging configurations for the project:
```json
+{
+  "version": "0.2.0",
+  "configurations": [...]
+}
```

Functionality in `app/api/chat/functions.ts` is expanded to include the `analyzeImage` function from the `./functions/analyzeImage.ts` file:
```typescript
+import analyzeImage from "./functions/analyzeImage";
+export const availableFunctions = {
+  analyzeImage,
...
}
```

A new `analyzeImage` module is created which lays the foundation for image analysis logic:
```typescript
+import OpenAI from "openai";
+...
+async function analyzeImage(...) {...}
+...
```

Routes in `app/api/chat/route.ts` have been modified to incorporate the new image analysis capabilities, removing prior handling logic for image messages and refactoring the `POST` method to integrate `analyzeImage` directly:
```typescript
-async function handleImageMessage(...) {...}
+import analyzeImage from "./functions/analyzeImage";
```

### :thinking: Surprising Changes
There are no obviously surprising changes in the diff; the changes are well-aligned with the stated intent of the PR to introduce image analysis functionality.

---
If you have any further details to discuss or require clarification, please feel free to add additional comments or reach out on Slack. Looking forward to seeing this feature in action! :rocket:

**Review by skylar-anderson — COMMENTED** [review]
Here's a comprehensive digest of the current PR:

**TL;DR**
- 	:toolbox: The PR aims to fix vision functionality in the system.
- 	:white_check_mark: It implements a new `analyzeImage` function within a function calling loop.
- 	:warning: Some concerns were raised about the max token limit for vision prompts in the discussion.
- 	:arrows_counterclockwise: Involved changes include addition of new files and significant code refactoring.

**Intent Summary**
The purpose of this pull request, which closes issue [#24](https://github.com/skylar-anderson/openai-chat-playground/issues/24), is to integrate image analysis capabilities with the existing codebase by adding an `analyzeImage` function. Previously, GPT-4 Vision calls were performed outside of a function calling loop due to limitations, which has since been revised to work within such a loop for better integration.

**Comments Summary**

:green_heart: [@skylar-anderson](https://github.com/skylar-anderson) said:
> This is super exciting! Can't wait to try it

:broken_heart: [@skylar-anderson](https://github.com/skylar-anderson) expressed concerns:
> I'm concerned that the max token limit may be too large for the vision prompt. Have you tested it?

**Open Questions**
- The user [@skylar-anderson](https://github.com/skylar-anderson) asked about the max token limit for the vision prompt and its testing, which remains unresolved in the comments.

**Change Walkthrough**
- A new `.vscode/launch.json` file has been added to the repository, enabling debugging configurations.
- Within `app/api/chat/functions.ts`, the `analyzeImage` function has been imported and added to the `availableFunctions` object, expanding the suite of functions the system can use.
- The `analyzeImage.ts` file defines the `analyzeImage` function, specifying its parameters and incorporating `gpt-4-vision-preview` for image analysis, requesting a detailed image description to aid in answering questions about the image.
- Changes to `app/api/chat/route.ts` involve a significant refactor, likely to accommodate the newly integrated vision capabilities of the system. The direct handling of image messages has been refactored to use the new function structure.
- The `Debugger/Column.tsx` component adjusts to no longer reference `imageAttached` property, aligning with the refactor focus on function calls.
- Frontend references to `base64File` have been removed in `app/page.tsx`, suggesting a change in how image data is managed or presented.
- `next.config.js` has simplifications, with lesser emphasis on experimental settings. Possibly indicative of a shift to stable features or a clean-up.

**Risky Changes**
- The introduction of the `gpt-4-vision-preview` model in `analyzeImage` function could be potentially risky if the max token limit is too large as raised by the comment from [@skylar-anderson](https://github.com/skylar-anderson). This needs to be tested thoroughly to avoid potential runtime or performance issues.
